src/main/java/com/microsoft/graph/concurrency/ChunkedUploadResponseHandler.java

```diff
@@ -154,30 +159,35 @@ public void configConnection(final Response response) {
             final ILogger logger) throws Exception {
         InputStream in = null;
         try {
-        	if (response.code() == HttpResponseCode.HTTP_ACCEPTED) {
-        		logger.logDebug("Chunk bytes has been accepted by the server.");
-        		in = new BufferedInputStream(response.body().byteStream());
-        		final UploadSession session = serializer.deserializeObject(
-        				DefaultHttpProvider.streamToString(in), UploadSession.class);
-
-        		return new ChunkedUploadResult<UploadType>(session);
-
-        	} else if (response.code() == HttpResponseCode.HTTP_CREATED
-        			|| response.code() == HttpResponseCode.HTTP_OK) {
-        		logger.logDebug("Upload session is completed, uploaded item returned.");
-        		in = new BufferedInputStream(response.body().byteStream());
-        		String rawJson = DefaultHttpProvider.streamToString(in);
-        		UploadType uploadedItem = serializer.deserializeObject(rawJson,
-        				this.deserializeTypeClass);
-
-        		return new ChunkedUploadResult<UploadType>(uploadedItem);
-        	} else if (response.code() >= HttpResponseCode.HTTP_CLIENT_ERROR) {
+        	if (response.code() >= HttpResponseCode.HTTP_CLIENT_ERROR) {
         		logger.logDebug("Receiving error during upload, see detail on result error");
 
         		return new ChunkedUploadResult<UploadType>(
         				GraphServiceException.createFromConnection(request, null, serializer,
         						response, logger));
-        	}
+            } else if (response.code() >= HttpResponseCode.HTTP_OK 
```

Comment:
The `if` accounts for all response codes 400 and higher. This `else if` accounts for all 200s. A final `else` should account for all 300s as being unexpected.

Reply:
there's already a default return at the end of the method. It was returning null but I added a more explicit error message.